### PR TITLE
27-07 edit

### PR DESCRIPTION
**Pull request by M-elhasan9**

### CODE REVIEW

File: app/Http/Controllers/Admin/CategoryCrudController.php [hunk 1]
```diff
@@ -29,6 +32,12 @@ public function setup()
         CRUD::setModel(\App\Models\Category::class);
         CRUD::setRoute(config('backpack.base.route_prefix') . '/category');
         CRUD::setEntityNameStrings('فئة', 'الفئات');
+
+        if (!backpack_user()->is_admin) {
```
Comment (OMR-prog) [inline app/Http/Controllers/Admin/CategoryCrudController.php:36]: only admin can add update and delete categories, he can not add materials

File: app/Http/Controllers/Admin/GroupCrudController.php [hunk 2]
```diff
@@ -76,4 +83,11 @@ protected function setupUpdateOperation()
     {
         $this->setupCreateOperation();
     }
+    protected function setupShowOperation()
```
Comment (OMR-prog) [inline app/Http/Controllers/Admin/GroupCrudController.php:86]: show materials that belongs to this group

File: app/Http/Requests/CategoryRequest.php [hunk 3]
```diff
@@ -26,7 +31,7 @@ public function authorize()
     public function rules()
     {
         return [
-            // 'name' => 'required|min:5|max:255'
+             'title' => 'required|min:3|max:255'
```
Comment (OMR-prog) [inline app/Http/Requests/CategoryRequest.php:34]: use unique constrain

File: app/Models/Category.php [hunk 4]
```diff
@@ -36,9 +36,10 @@ class Category extends Model
     | RELATIONS
     |--------------------------------------------------------------------------
     */
-    protected function materials()
+
+    public function materials()
     {
-        return $this->belongsToMany(Material::class,'material_category','category_id','material_id','id','id');
+        return $this->belongsToMany(Material::class,'category_material');
```
Comment (OMR-prog) [inline app/Models/Category.php:42]: both lines are correct, but first one is clearer, so I prefer it

File: app/Policies/MaterialPolicy.php [hunk 5]
```diff
@@ -30,6 +30,11 @@ public function materialToggleAvailability(User $user,Material $material)
         return $user->id == $material->user_id;
     }
 
+    public function authCheck()
+    {
+        return backpack_auth()->check();
```
Comment (OMR-prog) [inline app/Policies/MaterialPolicy.php:35]: what if we decide to allow non-logged users to see the app's materials?
the authorization responsibility falls only on policies

File: database/migrations/2021_07_27_062215_create_category_material_table.php [hunk 6]
```diff
@@ -13,7 +13,7 @@ class CreateMaterialCategoryTable extends Migration
      */
     public function up()
     {
-        Schema::create('material_category', function (Blueprint $table) {
+        Schema::create('category_material', function (Blueprint $table) {
```
Comment (OMR-prog) [inline database/migrations/2021_07_27_062215_create_category_material_table.php:16]: in laravel convention, you should write the tables names according to their order in migrations , so  the material comes before category

File: app/Http/Controllers/API/MaterialApiController.php [hunk 7]
```diff
@@ -24,12 +25,10 @@ public function index()
...
-    public function getCategory($categoryId){
-        $materials = DB::table('material_category')
-            ->where('category_id','=',$categoryId)
-            ->offset(0)->limit(10)->get('material_id');
-
-        return $materials;
+    public function getByCategory($categoryId){
+        //return all data of materials by category id
+        $categories = Category::find($categoryId);
+        return $categories->materials ;
```
Comment (OMR-prog) [inline app/Http/Controllers/API/MaterialApiController.php:31]: what about offset and limit?

File: app/Http/Controllers/API/MaterialApiController.php [hunk 8]
```diff
@@ -24,12 +25,10 @@ public function index()
...
 
-    public function getCategory($categoryId){
-        $materials = DB::table('material_category')
-            ->where('category_id','=',$categoryId)
-            ->offset(0)->limit(10)->get('material_id');
-
-        return $materials;
+    public function getByCategory($categoryId){
+        //return all data of materials by category id
+        $categories = Category::find($categoryId);
```
Comment (OMR-prog) [inline app/Http/Controllers/API/MaterialApiController.php:30]: two journies to the DB, try to make them one! 😊

File: app/Http/Controllers/API/MaterialApiController.php [hunk 9]
```diff
@@ -24,12 +25,10 @@ public function index()
         return $materials;
     }
 
-    public function getCategory($categoryId){
-        $materials = DB::table('material_category')
-            ->where('category_id','=',$categoryId)
-            ->offset(0)->limit(10)->get('material_id');
-
-        return $materials;
+    public function getByCategory($categoryId){
```
Comment (OMR-prog) [inline app/Http/Controllers/API/MaterialApiController.php:28]: where is the unit test?

File: app/Http/Requests/CategoryRequest.php (hunk 3 above)
Comment (M-elhasan9) [reply app/Http/Requests/CategoryRequest.php:34]: when i set the title to be unique i can not edit the title again

**Pull request merged**